internal/graphicscommand/image.go

```diff
@@ -190,25 +190,23 @@ func (i *Image) IsInvalidated() bool {
 	return i.image.IsInvalidated()
 }
 
-// Dump dumps the image to the specified path.
-// In the path, '*' is replaced with the image's ID.
+// DumpName formats the given pattern as a dump filename for the given image.
+// In the pattern, '*' is replaced with the image's ID.
+func (i *Image) DumpName(pattern string) string {
+	return strings.ReplaceAll(pattern, "*", strconv.Itoa(i.id))
+}
+
+// DumpTo dumps the image to the specified writer.
 //
 // If blackbg is true, any alpha values in the dumped image will be 255.
 //
 // This is for testing usage.
-func (i *Image) Dump(graphicsDriver graphicsdriver.Graphics, path string, blackbg bool, rect image.Rectangle) error {
+func (i *Image) DumpTo(graphicsDriver graphicsdriver.Graphics, blackbg bool, rect image.Rectangle, w io.Writer) error {
```

Comment:
1. We don't have to export this function, so `dumpTo` is fine.
2. I prefer `w` as the first argument.

Reply:
I think it needs to be exported, since it's being used in `internal/restorable/images_js.go`. `w` as the first argument makes sense though, will do.

Reply:
We should not expose how to dump the images to the outside, so you also don't have to add `DumpName`, and modify the restorable package.